pkg/agent/controller/traceflow/packetin.go

```diff
@@ -245,18 +247,58 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*crdv1alpha1.Tracefl
 				return nil, nil, nil, err
 			}
 		}
+		var pktMark uint32
+		if match := getMatchPktMarkField(matchers); match != nil {
+			pktMark, err = getMarkValue(match)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+		}
 		gatewayIP := c.nodeConfig.GatewayConfig.IPv4
 		if pktIn.Data.Ethertype == protocol.IPv6_MSG {
 			gatewayIP = c.nodeConfig.GatewayConfig.IPv6
 		}
 		gwPort := c.nodeConfig.GatewayConfig.OFPort
 		tunPort := c.nodeConfig.TunnelOFPort
 		if c.networkConfig.TrafficEncapMode.SupportsEncap() && outputPort == tunPort {
+			egressPacket := c.egressQuerier.EgressPacket(tunnelDstIP)
+			if egressPacket { // an Egress packet, currently on Source Node and forwarded to Egress Node.
+				obEgress := new(crdv1alpha1.Observation)
+				obEgress.EgressIP = tunnelDstIP
+				egressPolicy, err := c.egressQuerier.GetEgressPolicy(ns, srcPod)
```

Comment:
what if ns and srcPod is empty (live traffic mode)? the code would fail the whole traceflow I guess.

Reply:
Can destination be a Pod in the case of Egress ?

Reply:
No, Egress means it's accessing external network, please check https://github.com/antrea-io/antrea/blob/main/docs/egress.md#what-is-egress

Reply:
Yes, so this case would never occur because live traffic requires atleast one out of source and destination to be Pod and in Egress case destination cannot be a Pod, so srcPod and ns can't be empty.

Reply:
You are correct if the packet is truely Egress. However, the above check `IsEgressIP` is not always correct. When using this mode https://github.com/antrea-io/antrea/blob/main/docs/egress.md#configuring-static-egress. A normal node IP can be used as egress IP, so it could happen the traffic is towards a Pod on another Node but is considered as Egress.

Reply:
Understood.